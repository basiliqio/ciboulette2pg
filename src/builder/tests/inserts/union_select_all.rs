use super::*;

#[test]
fn multi() {
    let mut builder = Ciboulette2PostgresBuilder::default();
    let dest_table = Ciboulette2PostgresTableSettings::new(
        Ciboulette2PostgresSafeIdent::try_from("id").unwrap(),
        Ciboulette2PostgresSafeIdent::try_from("uuid").unwrap(),
        Some(Ciboulette2PostgresSafeIdent::try_from("public").unwrap()),
        Ciboulette2PostgresSafeIdent::try_from("peoples").unwrap(),
    );

    let from_table = Ciboulette2PostgresTableSettings::new(
        Ciboulette2PostgresSafeIdent::try_from("id").unwrap(),
        Ciboulette2PostgresSafeIdent::try_from("uuid").unwrap(),
        Some(Ciboulette2PostgresSafeIdent::try_from("public").unwrap()),
        Ciboulette2PostgresSafeIdent::try_from("articles").unwrap(),
    );
    builder
        .included_tables
        .insert(&from_table, from_table.clone());
    builder
        .included_tables
        .insert(&dest_table, dest_table.clone());
    builder.gen_union_select_all().unwrap();
    let res = builder.build().unwrap();
    insta::assert_debug_snapshot!(res);
}

#[test]
fn single() {
    let mut builder = Ciboulette2PostgresBuilder::default();
    let dest_table = Ciboulette2PostgresTableSettings::new(
        Ciboulette2PostgresSafeIdent::try_from("id").unwrap(),
        Ciboulette2PostgresSafeIdent::try_from("uuid").unwrap(),
        Some(Ciboulette2PostgresSafeIdent::try_from("public").unwrap()),
        Ciboulette2PostgresSafeIdent::try_from("peoples").unwrap(),
    );
    builder
        .included_tables
        .insert(&dest_table, dest_table.clone());
    builder.gen_union_select_all().unwrap();
    let res = builder.build().unwrap();
    insta::assert_debug_snapshot!(res);
}

#[test]
fn no_table() {
    let mut builder = Ciboulette2PostgresBuilder::default();
    builder.gen_union_select_all().unwrap();
    let res = builder.build().unwrap();
    insta::assert_debug_snapshot!(res);
}
