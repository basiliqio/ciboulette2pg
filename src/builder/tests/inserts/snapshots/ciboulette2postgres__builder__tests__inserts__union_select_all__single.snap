---
source: src/builder/tests/inserts/union_select_all.rs
expression: "sqlformat::format(query.as_str(), &sqlformat::QueryParams::None,\n                  sqlformat::FormatOptions::default())"

---
SELECT
  "public"."peoples"."id",
  "public"."peoples"."type",
  "public"."peoples"."data"
FROM
  "public"."peoples";
